lib/ddtrace/contrib/action_view/instrumentation.rb

```diff
@@ -0,0 +1,192 @@
+require 'ddtrace/contrib/action_view/ext'
+
+module Datadog
+  module Contrib
+    module ActionView
+      # Defines instrumentation for ActionView
+      module Instrumentation
+        # Instrumentation for template rendering
+        module TemplateRenderer
+          # Rails < 3.1 template rendering
+          module Rails30
+            # rubocop:disable Metrics/MethodLength
+            def self.prepended(base)
+              # rubocop:disable Metrics/BlockLength
+              base.class_eval do
+                def render_with_datadog(*args, &block)
+                  # NOTE: This check exists purely for Rails 3.0 compatibility.
+                  #       The 'if' part can be removed when support for Rails 3.0 is removed.
+                  if active_datadog_span
```

Comment:
Not sure I fully follow this logic. I am sure it makes sense, just I think I am missing something.

Reply:
e.g. why do we need to do this?